tools/src/main/java/org/neo4j/tools/txlog/CheckTxLogs.java

```diff
@@ -121,21 +134,30 @@ void scan( File[] logs, InconsistenciesHandler handler, CheckType<?,?>... checkT
         }
     }
 
-    private <C extends Command, R extends Abstract64BitRecord> void process( C command, CheckType<C,R> check,
+    private <C extends Command, R extends Abstract64BitRecord> boolean checkAndHandleInconsistencies( C command, CheckType<C,R> check,
             CommittedRecords<R> state, long logVersion, InconsistenciesHandler handler )
     {
+        boolean inconsistent;
+
         R before = check.before( command );
         R after = check.after( command );
 
-        if ( !state.isValid( before ) )
-        {
-            LogRecord<R> seen = state.get( before.getId() );
-            LogRecord<R> current = new LogRecord<>( before, logVersion );
```

Comment:
The logic was rewritten so that it is clearer what the check does. Everything operates around after/before so it seemed clearer to not abstract that away. The method as a whole reads clearer now.